swri_transform_util/src/nodes/obstacle_transformer.cpp

```diff
@@ -0,0 +1,88 @@
+// *****************************************************************************
+//
+// Copyright (C) 2019 All Right Reserved, Southwest Research Institute® (SwRI®)
+//
+// THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
+// KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
+// PARTICULAR PURPOSE.
+//
+// *****************************************************************************
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <marti_nav_msgs/msg/obstacle.hpp>
+#include <marti_nav_msgs/msg/obstacle_array.hpp>
+
+#include <swri_roscpp/publisher.h>
+#include <swri_roscpp/subscriber.h>
+
+#include <swri_transform_util/transform_manager.h>
+
+namespace swri_transform_util
+{
+class ObstacleTransformer : public rclcpp::Node
+  {
+  public:
+    explicit ObstacleTransformer(const rclcpp::NodeOptions& options) :
+      rclcpp::Node("obstacle_transformer", options),
+      tf_manager_(shared_from_this())
+    {
+      this->declare_parameter("output_frame", "/wgs84");
+
+      output_frame_ = this->get_parameter("output_frame").as_string();
+
+      object_array_sub_ = swri::Subscriber(*this,
+                                           "object_array",
+                                           1,
+                                           &ObstacleTransformer::handleObstacleArray,
+                                           this);
+
+      viz_pub_ = swri::advertise<marti_nav_msgs::msg::ObstacleArray>(*this, "viz_array", 1);
+    }
+
+  private:
+    void handleObstacleArray(const marti_nav_msgs::msg::ObstacleArray::ConstSharedPtr& obj)
+    {
+      if (viz_pub_->get_subscription_count() == 0 && viz_pub_->get_intra_process_subscription_count() == 0)
```

Comment:
Wow, does subscription count really not include the intraprocess subscriptions?

Reply:
Nope: https://github.com/ros2/rclcpp/issues/668

Reply:
That's annoying